Vendor Company Engagement

v0lkan opened the pull request:
This PR outlines guidelines about how SasS companies can collaborate with the SPIFFE community while maintaining a healthy, mutual, and respectful relationship.

v0lkan committed:
Initial draft for SasS company engagerment

Signed-off-by: Volkan Özçelik <volkan.ozcelik@broadcom.com>

v0lkan commented:
Still working on it… but it’s in a good enough shape to receive some eyeballs :)

amoore877 commented on community/saas-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,118 @@
+
+##### Designation of Contacts
+
+Each SaaS company that directs customers to the SPIFFE Slack channel must 
```
```suggestion
Each SaaS company that directs their customers to the SPIFFE Slack channel must 
```
just to be very specific

v0lkan replied:
yup makes sense.

amoore877 commented on community/saas-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,118 @@
+designate two technical contacts.
+
+These contacts will serve as the primary point of contact for addressing queries 
+and facilitating communication between the customers and the Spiffy community.
```
```suggestion
and facilitating communication between the customers and the SPIFFE community.
```

kfox1111 replied:
And provide a mechanism for the SPIFFE community to communicate with the Company?

v0lkan replied:
There are several of those Spiffy’s. side effect of multitasking :) — will fix them all.

amoore877 commented on community/saas-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,118 @@
+The folder will contain:
+
+* Names and contact information of the designated technical contacts.
+* Guidelines for interaction and support procedures.
```
```suggestion
* Company-specific guidelines for interaction and support procedures.
```

amoore877 commented on community/saas-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,118 @@
+
+##### Communication Etiquette
+
+All interactions should be professional, respectful, and focused on technical 
```
```suggestion
All interactions must be professional, respectful, and focused on technical 
```

v0lkan replied:
That makes sense too.

Btw, I may convert some of those paragraphs to bullet points, not sure. — but will keep the wording mostly intact (aside from y'all's feedback of course.

amoore877 commented on community/saas-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,118 @@
+##### Communication Etiquette
+
+All interactions should be professional, respectful, and focused on technical 
+issues.
```
```suggestion
issues. The Code of Conduct is [here](../CODE_OF_CONDUCT.md).
```

v0lkan replied:
Noted.

amoore877 commented on community/saas-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,118 @@
+Disruptive behavior includes, but is not limited to, spamming, harassment, and 
+consistently off-topic discussions.
+
+Technical contacts should monitor interactions and identify any disruptive 
```
```suggestion
Technical contacts must monitor interactions and identify any disruptive 
```

amoore877 commented on community/saas-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,118 @@
+The SSC will evaluate the situation and decide on appropriate actions, which may 
+include temporary or permanent removal from the Slack channel.
+
+##### Documentation and Reporting
```
```suggestion
include temporary or permanent removal from the Slack channel.

If a particular company's customers have a mass habit of disruption or other issues, they will be evicted from channels and the list of community partners.

##### Documentation and Reporting
```

v0lkan replied:
Makes sense, maybe I'll alter the wording a bit. 
Will update and let here know.

strideynet commented on community/saas-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,118 @@
+The folder will contain:
+
+* Names and contact information of the designated technical contacts.
+* Guidelines for interaction and support procedures.
```
It's not clear to me what "Guidelines for interaction" actually involves. Is this simply a case of pointing folks towards the appropriate ticket portal on the SAAS provider side?

v0lkan replied:
I believe it will be up to SSC and Company to agree upon. 

These could include steps for submitting tickets, expected response times, preferred communication channels, and any other relevant interaction policies.

Thanks for the feedback, I’ll clarify that part a bit.

v0lkan replied:
@strideynet (regarding the above longer comment)

Thank you for sharing your thoughts and concerns. I understand and appreciate the challenges you've highlighted regarding vendors' responsibility in moderating community behavior, especially when it involves individuals without a direct commercial relationship.

Our aim with these guidelines is to ensure a positive and productive community environment. However, we also do recognize the difficulties of moderation and the potential burden it may place on technical contacts.

Humans are nondeterministic beings. And that makes all human-related issues convoluted.

Given the complexities you've mentioned, discussing this further within the SPIFFE Steering Committee (SSC) would be beneficial. We want to ensure that our policies are practical and do not discourage valuable contributions to the community.

Your feedback is invaluable as we continue to refine our guidelines.

Thanks again all your comments and feedback.

strideynet replied:
> Humans are nondeterministic beings. And that makes all human-related issues convoluted.

Never a truer thing said :")

I appreciate you taking the feedback into account - I'll admit that I've not had much of an opportunity to give back to the SPIFFE community yet but it's certainly in my plans for this year.

---

I think if I were to expand a little on my thoughts - based on this document, it would appear that there's two broad goals with the proposed vendor engagement strategy:

- Technical: Setting out ways for the SPIFFE team to get in contact with technical folks at these vendor projects, for the purposes of technical discussion etc.
- Social: Setting out a level of commitment that that vendor needs to make to fostering a positive and growing SPIFFE community, and engaging in the work of trying to moderate it.

Both of these are good goals and I think it's a reasonable expectation of vendor to "give back" in both these ways.

However, I'm not sure bundling the second element with the first necessarily makes sense. Perhaps some of the time it does - I'd be happy to engage with both elements on behalf of Teleport - but when it comes to picking the "second" contact, I face a choice:

- Do I ask someone who is much stronger at community engagement and moderation, but weaker when it comes to the technical details of our SPIFFE implementation? (e.g a member of our devrel team)
- Do I ask someone who is much stronger at the technical details of our SPIFFE implementation but weaker when it comes to community engagement? (e.g a member of my engineering team)

I don't think this "trade off" is specific to us, I've seen plenty of similar orgs where the skills needed here are broken up into different parts of the organisation. 

What I have in mind is perhaps breaking up these roles:

- Technical Contacts
  - Intended to be individuals with a high level of context on the implementation of the vendor's SPIFFE product.
  - Role: contact point for SPIFFE contributors who have questions about the details of a vendor implementation, and for escalating technical queries.
- Community Contacts
  - Intended to be individuals who perform devrel/community work for the vendor.
  - Role: Engage with moderation side.
  - Role: good contact point for discussing content which could benefit both the vendor and the SPIFFE project as a whole (e.g publicising case studies, use case statements, blog posts)

I think it wouldn't be a problem to have an individual be both a Technical Contact and a Community Contact but by breaking up the roles we can make this more flexible and potentially more accessible. I can imagine individuals who'd be keen to be more involved with the relationship with SPIFFE, but might be hesitant to take on both technical and social aspects.

strideynet commented on community/saas-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,118 @@
+All interactions should be professional, respectful, and focused on technical 
+issues.
+
+Technical contacts are responsible for ensuring that their customers, and anyone
```
I appreciate the sentiment here, but it's a little unclear what the extent of this responsibility is. Whilst I wouldn't knowingly direct a customer who would act contrary to the guidelines towards this slack, in the case of generally linking to the SPIFFE community within documentation and such, it's difficult for a vendor to control/moderate the actions of those who follow a link in public documentation.

strideynet commented on community/saas-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,118 @@
+
+The folder will contain:
+
+* Names and contact information of the designated technical contacts.
```
It might be helpful here to define more specifically what "contact information" here needs to be. Can we provide a group email (e.g spiffe-community@goteleport.com) that can be directed to the appropriate contacts - or must this be specific contact details for these individuals?

v0lkan replied:
I think both individual contact email, or a distribution list would work — We’ll of course discuss this further within SSC. — and at any rate, nothing is set in stone, we can always amend this.

Thanks for all your insightful comments btw. Truly appreciate it.

strideynet commented on community/saas-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,118 @@
+Disruptive behavior includes, but is not limited to, spamming, harassment, and 
+consistently off-topic discussions.
+
+Technical contacts should monitor interactions and identify any disruptive 
```
If we identify that one of our customers is responsible for disruptive behaviour, what actions should we take? Should we determine that this behaviour breaches the conduct guidelines ourself and reach out directly to the customer, or should we we to escalate this to the SPIFFE team first?

In the case that we don't identify this behaviour before SSC do, should we expect to be contacted to deliver the warning?

v0lkan replied:
Here's what I have in mind:

* If a technical contact identifies disruptive behavior, they should first determine * if this behavior breaches the conduct guidelines.
* Upon confirming a breach, the technical contact should reach out directly to the customer to deliver a warning and outline the expected conduct.
* If the behavior persists after the warning, they should escalate the issue to the SPIFFE Steering Committee (SSC).



I’ll amend the document too.

strideynet commented on community/saas-companies-engagement-procedure.md:
Is it worth reframing "SaaS" as "vendor" here? At least personally, "SaaS" tends to refer to "hosted" products, whereas "vendor" would likely refer to any commercial entity interacting in the SPIFFE space regardless of whether the software is hosted by the vendor or the customer.

v0lkan replied:
Not sure.

@amoore877 , what do you think?

strideynet reviewed (commented):
I appreciate the overall sentiment here - it makes sense to put in some guidelines as to how vendors interact with the community. 

However, I think I have some general concerns as to how holding vendors responsible for the behaviour of those that they've referred to the community would actually work. I'm keen to link out to the SPIFFE community within our documentation etc but, it's challenging for me to "vet" those who follow a link, especially if this may be individuals to whom we do not have a commercial relation (e.g those who use our open source product).

I understand there's plenty of challenges that come with community moderation, but I'm not sure that asking vendors to personally moderate those who've discovered SPIFFE through them is necessarily the right solution. Moderation is difficult - and I'm not sure that those who might be considered "technical contacts" are necessarily the right people to carry this out.

It's understandable if this is the path that SPIFFE wishes to take, but I worry that the outcome of this would be that vendors are reluctant to link to the SPIFFE community.

v0lkan committed:
Addressing comments

Signed-off-by: Volkan Özçelik <volkan.ozcelik@broadcom.com>

v0lkan committed:
Merge branch 'main' into ovolkan/sass

v0lkan committed:
Rename “sass” as vendor

Signed-off-by: Volkan Özçelik <volkan.ozcelik@broadcom.com>

v0lkan committed:
expand and clarify kinds of contacts

Signed-off-by: Volkan Özçelik <volkan.ozcelik@broadcom.com>

v0lkan commented:
I believe I addressed all the comments. 

Also as per @strideynet’s suggestion, I expanded the definition of what kinds of contacts there can be and what their corresponding roles are.

I will ping SPIFFE Slack for further reviews and comments.

v0lkan committed:
add channel removal procedure

Signed-off-by: Volkan Özçelik <volkan.ozcelik@broadcom.com>

kfox1111 reviewed (dismissed):
LGTM. Thanks! :)

amoore877 commented on community/slack-communication-guidelines.md:
```diff
@@ -0,0 +1,109 @@
+#### Follow the Escalation Process
+
+If disruptive behavior continues, follow the escalation process outlined in the
+[Procedure for Engaging With SaaS Companies](saas-companies-engagement-procedure.md)
```
link needs correction now

amoore877 commented on community/vendor-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,185 @@
+##### Documentation of Contacts
+
+A dedicated folder for each company will be created in the SPIFFE repository
+under the `./community/saas-technical-contacts` directory.
```
folder name probably should be updated to match name change

v0lkan committed:
Merge branch 'main' into ovolkan/sass

v0lkan committed:
minor wording changes

Signed-off-by: Volkan Özçelik <volkan.ozcelik@broadcom.com>

kfox1111 commented on community/slack-communication-guidelines.md:
```diff
@@ -0,0 +1,109 @@
+* Severe or repeated violations may result in permanent removal from the Slack 
+channel.
+* Decisions for permanent removal will be made by the SSC team and communicated
+  the interested parties.
```
```suggestion
  to the interested parties.
```

kfox1111 commented on community/vendor-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,184 @@
+  must designate two contacts. At least one of these contacts are encouraged
+  (*but not required*) to be a technical contact. See the definition of roles
+  and responsibilities of technical and community contacts in the subsequent
+  **Kinds of Contacts** subsection.
```
Missing blank line between paragraphs?
```suggestion
  **Kinds of Contacts** subsection.
  
```

v0lkan replied:
It's a (bit long…ish) bullet point, not a paragraph. Adding a new line would make it two separate bullet points with one item on each.

kfox1111 commented on community/vendor-companies-engagement-procedure.md:
```diff
@@ -0,0 +1,184 @@
+* All interactions must be professional, respectful, and focused on technical 
+  issues. [Refer to the **SPIFFE Code of Conduct** for details][coc].
+* Technical contacts are responsible for ensuring that their customers, and 
+  anyone who join the community through their reference adhere to the community 
```
```suggestion
  anyone who joins the community through their reference adhere to the community 
```

kfox1111 reviewed (commented):
Looking great. A few minor formatting/grammar related suggestions.

v0lkan committed:
minor typo fixes

Signed-off-by: Volkan Özçelik <volkan.ozcelik@broadcom.com>

kfox1111 reviewed (approved):
LGTM. Thanks! :)

amoore877 commented:
should probably add this folder to the CODEOWNERS as under SSC ownership

v0lkan commented:
> should probably add this folder to the CODEOWNERS as under SSC ownership

I can add that as part of this PR too — that, too, I think, will require a CODEOWNER approval :) .

v0lkan commented:
I'm confused:

The `/community` folder and any subfolder is owned by SSC; however, it still requires an approving review.

@kfox1111 is from the SSC, and he approved it; along with @amoore877 .

v0lkan commented:
Oh shoot! I touched `.gitignore`, that's why :D .

@evan2645 @justinburke @spikecurtis @azdagron : can you check and approve this PR when you have time; thanks a lot.

azdagron reviewed (approved):
Approved the .gitignore addition
